stgpytools/types/file.py

```diff
@@ -134,5 +138,58 @@ def move_dir(self, dst: SPath, *, mode: int = 0o777) -> None:
 
         self.rmdir()
 
+    def lglob(self, pattern: str = '*') -> list[SPath]:
+        """Glob the path and return the list of paths."""
+
+        return [SPath(p) for p in self.glob(pattern)]
+
+    def fglob(self, pattern: str = '*') -> SPath | None:
+        """Glob the path and return the first match."""
+
+        for root, dirs, files in walk(self):
+            for name in dirs + files:
+                if fnmatch.fnmatch(name, pattern):
+                    return SPath(path.join(root, name))
+
+        return None
+
+    def find_newest_file(self, pattern: str = '*') -> SPath | None:
+        """Find the most recently modified file matching the given pattern in the directory."""
+
+        matching_files = self.glob(pattern)
+
+        if not matching_files:
+            return None
+
+        return max(matching_files, key=lambda p: p.stat().st_mtime, default=None)  # type:ignore
+
+    def is_empty_dir(self) -> bool:
+        """Check if the directory is empty."""
+
+        return self.is_dir() and not any(self.iterdir())
+
+    def get_size(self) -> int:
+        """Get the size of the file or directory in bytes."""
+
+        if self.is_file():
+            return self.stat().st_size
+        elif self.is_dir():
+            return sum(f.stat().st_size for f in self.rglob('*') if f.is_file())
```

Comment:
so this only works for top-level files? does rglob used like this traverse dirs?

Reply:
What does the r stand for

Reply:
😁😃😄